src/main/java/io/pravega/connectors/flink/Pravega.java

```diff
@@ -253,6 +280,24 @@ protected TableSourceReaderBuilder withDeserializationSchema(DeserializationSche
             return this;
         }
 
+        /**
+         * Configures the timestamp and watermark assigner.
+         *
+         * @param assignerWithTimeWindows the timestamp and watermark assigner.
+         * @return TableSourceReaderBuilder instance.
+         */
+        // TODO: Due to the serialization validation for `connectorProperties`, only `public` `static-inner/outer` class implements
+        // `AssignerWithTimeWindow` is supported as a parameter of `withTimestampAssigner` in Table API stream table source.
+        protected TableSourceReaderBuilder withTimestampAssigner(AssignerWithTimeWindows<Row> assignerWithTimeWindows) {
+            try {
+                ClosureCleaner.clean(assignerWithTimeWindows, true);
+                this.assignerWithTimeWindows = new SerializedValue<>(assignerWithTimeWindows);
+            } catch (IOException e) {
```

Comment:
`ClosureClean` will throw an `InvalidProgramException` if the object fails serialization and not `IOException` (which is caught and transformed by `ClosureClean`.

`InvalidProgramException` does not extend `IOException` and therefore this will not catch the Exception